### ISSUE TITLE
showReportDialog() gives "Bad data decoding request"

### ISSUE THREAD

**Issue by hatton** [getsentry/sentry-electron#129]
using @sentry/electron": "^0.11.0",  I initialize sentry electron with

```javascript

const { init, showReportDialog } = require("@sentry/electron");

exports.initializeSentry = () => {
  init({
    dsn: <my dsn>,
    enableNative: false,
    release: require("./package.json").version,
    beforeSend(event) {
      if (event.exception) {
        showReportDialog();
      } else {
        return event;
      }
    }
  });
};
```

This is called both from the main process and the render process. The docs tell me  the beforeSend is a noop in the main process. OK, so I trigger an error either with a raw throw or an explicit call to ``Sentry.captureException(e)``. The box comes up but before I enter anything in the dialog box, I see this error in the terminal:

```
{ SentryError: HTTP Error (400): Bad data decoding request (TypeError, Incorrect padding)
    at new SentryError (C:\dev\saymore-js\node_modules\@sentry\core\dist\error.js:21:28)
    at ClientRequest.req.on (C:\dev\saymore-js\node_modules\@sentry\electron\dist\main\transports\net.js:44:36)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at URLRequest.ClientRequest.urlRequest.on (C:\dev\saymore-js\node_modules\electron\dist\resources\electron.asar\browser\api\net.js:207:12)
    at emitOne (events.js:116:13)
    at URLRequest.emit (events.js:211:7) name: 'SentryError' }
```

Seeing this in the terminal meant that it's coming from main process. Sure enough, if I don't initialize sentry from the main process, I don't see this error, either in the terminal or the chrome debug console. However, I assume it is still breaking, because I've never seen the results of my feedback show up in Sentry, and as far as I can tell, the event itself does not show up if I have this ``beforeSend()`` parameter. If I remove it, then I get events.

**Comment by hatton** [comment]
Found a ``debug`` option, and turning that on gave me this error:

``Sentry Logger [Error]: `beforeSend` method has to return `null` or a valid event``

which makes sense. so I changed the code to:

```javascript
    beforeSend(event) {
      if (event.exception) {
        showReportDialog();
      }
      return event;
    }
```

Which made that error go away but didn't change the behavior, otherwise. There are no other errors showing as a result of the ``debug`` option.

**Comment by hatton** [comment]
Tried with the new release, 0.12.0, no change.

**Comment by timfish** [comment]
We have just released v3.0.0 of the SDK which includes a lot of changes and improvements since this issue was opened. 

If you can still reproduce this with v3, please open a new issue and I'll look into it.
